caravel/db_engines.py

```diff
@@ -0,0 +1,155 @@
+"""Compatibility layer for different database engines
+
+This modules stores logic specific to different database engines. Things
+like time-related functions that are similar but not identical, or
+information as to expose certain features or not and how to expose them.
+
+For instance, Hive/Presto supports partitions and have a specific API to
+list partitions. Other databases like Vertica also support partitions but
+have different API to get to them. Other databases don't support partitions
+at all. The classes here will use a common interface to specify all this.
+
+The general idea is to use static classes and an inheritance scheme.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import inspect
+from collections import namedtuple
+from flask_babel import lazy_gettext as _
+
+Grain = namedtuple('Grain', 'name label function')
+
+
+class BaseEngine(object):
+    engine = 'base'  # str as defined in sqlalchemy.engine.engine
+    epoch_to_dttm = None
+
```

Comment:
> str as defined in sqlalchemy.engine.engine
I guess its sqlalchemy.engine.Engine.name

Describe the interface BaseEngine completely by adding
` time_grains = () #Tuple of Grain. Time grains supported by the database. `

Besides defining the used interface properly, it prevents the crash in model.Database.grain/grain_dict in the case of 'base' which is used if the SQL dialect is not listed here.